### PR TITLE
feat(depinject): support 08 wasm

### PR DESCRIPTION
**Pull request by DimitrisJim**
## Description

ref: #3560

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/light-clients/08-wasm/depinject.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+// ProvideModule returns the 08-wasm module outputs for dependency injection
+func ProvideModule(in ModuleInputs) ModuleOutputs {
+	// default to governance authority if not provided
+	authority := authtypes.NewModuleAddress(govtypes.ModuleName)
+	if in.Config.Authority != "" {
+		authority = authtypes.NewModuleAddressOrBech32Address(in.Config.Authority)
+	}
+
+	keeper := wasmkeeper.NewKeeperWithVM(
```
[the remark this answers is not in the document]
Comment (damiannolan) [reply modules/light-clients/08-wasm/depinject.go:59]: Can check some more in depth later today, we can also bring this up tomorrow when syncing with the sdk team.

File: modules/light-clients/08-wasm/depinject.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+		authority = authtypes.NewModuleAddressOrBech32Address(in.Config.Authority)
+	}
+
+	var keeper wasmkeeper.Keeper
+	if in.VM != nil {
+		keeper = wasmkeeper.NewKeeperWithVM(
+			in.Cdc, in.StoreService, in.ClientKeeper, authority.String(), in.VM, in.QueryRouter, in.Opts...,
+		)
+	} else {
+		// TODO(jim): If missing, its default value is used. This could very well be surprising and cause misconfiguration
```
[the remark this answers is not in the document]
Comment (crodriguezvega) [reply modules/light-clients/08-wasm/depinject.go:68]: If `dataDir` is `""` maybe an error is already returned in the call to `wasmvm.NewVM`?

**Review by crodriguezvega — COMMENTED** [review]
Looks good to me, fine merging this to move forward and come back to it if needed later on.

**Review by crodriguezvega — APPROVED** [review]

**Comment by crodriguezvega** [comment]
I am going to merge this for now, any improvements can be handled in follow up PRs.

**Pull request merged**